hl7/4.2/modules/ROOT/edits/hl7-connector-config-topics.adoc

```diff
@@ -0,0 +1,269 @@
+
+The possible number of occurrences of a segment or group is defined by the
+count value. This defaults to a value of `1`.
+
+To delete segments from the message structure or change segment requirements from mandatory to optional (or vice versa), make the change in your copy of the schema and use the modified version in your application.
+
+You can add segments that are not present in the original message
+structure definition. First remove the position values from all existing segment and group definition lines in the schema, otherwise you need to renumber everything following an added segment.
+If you remove the explicit position numbers, segments and groups are assigned
+position numbers sequentially, and for most purposes these numbers are not
+seen by HL7 EDI Connector applications.
+
+If you want to add a standard HL7 segment to a message structure,
+reference it with an `idRef` line at the appropriate place, and HL7 obtains
+the definition from the `basedefs.esl` file referenced as an import.
+
+The following partial example shows `CON` segments added to an ADT_A05 message structure:
+
+[source,yaml,linenums]
+----
+form: HL7
+version: '2.5.1'
+imports: [ '/hl7/v2_5_1/basedefs.esl' ]
+id: 'ADT_A05'
+name: 'ADT_A05'
+data:
+- { idRef: 'MSH', usage: M }
+- { idRef: 'SFT', usage: O, count: '>1' }
+- { idRef: 'EVN', usage: M }
+- { idRef: 'PID', usage: M }
+- { idRef: 'PD1', usage: O }
+- { idRef: 'CON', usage: O, count: '>1' }
+- { idRef: 'NTE', usage: O }
+- { idRef: 'ROL', usage: O, count: '>1' }
+- { idRef: 'NK1', usage: O, count: '>1' }
+- { idRef: 'PV1', usage: M }
+- { idRef: 'PV2', usage: O }
+- { idRef: 'CON', usage: O, count: '>1' }
+- { idRef: 'ROL', usage: O, count: '>1' }
+...
+----
+
+If you want to define a non-standard segment for your message structure,
+add the segment definition to the schema. This is more complex than just
+modifying the segment structure, as you need to list all components
+in the segment. The easiest starting point for this is to find a similar
+standard HL7 segment and copy the definition used for that standard
+segment from the `basedefs.esl` file. You can then add the segments key
+following your message structure definition, followed by one or more
+segment definitions.
+
+=== Set Your Schema Locations
+
+You can configure schema locations in the Anypoint Studio XML view.
+
+. Switch to the XML view by clicking *Configuration XML*.
+
+. Modify your HL7 EDI configuration to include a list of all the schemas to include by adding an `+<http://edischema[edi:schema]>+` element for each document type:
++
+[source,xml,linenums]
+----
+<hl7-edi:config name="HL7_EDI__Configuration" identKeys="true" doc:name="HL7 EDI: Configuration">
+  <hl7-edi:schemas>
+    <hl7-edi:schema>hl7/v2_6/ADT_A01.esl</hl7-edi:schema>
+  </hl7-edi:schemas>
+</hl7-edi:config>
+----
++
+. After you create a global element for your HL7 EDI, configure the message structure, operations, and acknowledgments.
```

Comment:
This led me to expect separate subtopics about configuration, operation, and acknowledgement that I do not see following....

Reply:
Fixed.